frontend/src/features/SidebarSuite/uiSettings/config.ts

```diff
@@ -0,0 +1,114 @@
+import { DbLanguage } from "@utils/api/types";
+import { exhaustiveStringTuple } from "@utils/helpers";
+import {
+  AllUIComponentParamNames,
+  DBSourceFilePageFilterUISettingName,
+  DisplayUISettingName,
+  RequestFilterName,
+  SearchPageFilterUISettingName,
+  SortMethod,
+  UtilityUISettingName,
+} from "src/features/SidebarSuite/types";
+
+export const allRequestFilters = exhaustiveStringTuple<RequestFilterName>()(
+  "languages",
+  "include_files",
+  "exclude_files",
+  "include_categories",
+  "exclude_categories",
+  "include_collections",
+  "exclude_collections",
+  "par_length",
+  "score",
+  "language",
+);
+
+// order sets appearance in sidebar
+
+export const dbSourceFileRequestFilters =
+  exhaustiveStringTuple<DBSourceFilePageFilterUISettingName>()(
+    "score",
+    "par_length",
+    "exclude_sources",
+    "include_sources",
+    "languages",
+  );
+
+export const searchRequestFilters =
+  exhaustiveStringTuple<SearchPageFilterUISettingName>()(
+    "language",
+    "exclude_sources",
+    "include_sources",
+  );
+
+export const displayUISettings = exhaustiveStringTuple<DisplayUISettingName>()(
+  "folio",
+  "sort_method",
+  "script",
+  "showSegmentNrs",
+);
+
+export const sortMethods = exhaustiveStringTuple<SortMethod>()(
+  "position",
+  "quotedtext",
+  "length",
+  "length2",
+);
+
+export const utilityUISettings = exhaustiveStringTuple<UtilityUISettingName>()(
+  "download_data",
+  "copyQueryTitle",
+  "copyQueryLink",
+  "emailQueryLink",
+);
+
+export const allUIComponentParamNames: AllUIComponentParamNames = {
```

Comment:
is this needed? Looks like the keys and values are exactly the same.

Reply:
Necessity, preference... who's to say :shrug: :smile: 

The degree of benefit might be small as it's not actually used in many places, but I still there is use in having centralized values rather than using literal strings through the code base. It is effectively serving the role of an enum except is based on an api derived type.